haskell/src/Lentille.hs

```diff
@@ -94,6 +106,7 @@ instance MonadLog LentilleM where
 
 instance MonadRetry LentilleM where
   retry = retry'
+  genericRetry = genericRetry'
```

Comment:
would have been nice if retry was implemented in term of genericRetry so that the MonadRetry only need a single method. But that's ok like that too.

Reply:
Yes true, I'll let a TODO note for that. An awesome task for a rainy Sunday .... ;)

Reply:
Well the retry class is really about an abstraction for the `retry` library, e.g. to be able to catch and re-throw exception. Thus I think we should, on a rainy Sunday, try to implement the http retry in term of genericRetry so that the class only has a single retry function. But that's more of a nit, it's ok like what you proposed.